tests/meson.build

```diff
@@ -0,0 +1,86 @@
+tests = [
+  {
+    'name': 'test-brush-load',
+  },
+  {
+    'name': 'test-brush-persistence',
+  },
+  {
+    'name': 'test-details',
+  },
+  {
+    'name': 'test-fixed-tiled-surface',
+  },
+  {
+    'name': 'test-rng',
+  },
+]
+
+if use_gegl
+  tests += {
+    'name': 'test-gegl-surface',
+    'srcs': 'gegl/test-gegl-surface.c',
+    'deps': [
+      gegl,
+    ],
+    'incs': [
+      libmypaint_gegl_inc,
+    ],
+    'link': [
+      libmypaint_gegl,
+    ],
+  }
+endif
+
+libmypaint_tests_lib = static_library(
+  'mypaint-tests',
+  'mypaint-benchmark.c',
+  'mypaint-test-surface.c',
+  'mypaint-utils-stroke-player.c',
+  'testutils.c',
+  brush_settings_headers,
+  c_args: [
+    '-DLIBMYPAINT_TESTING_ABS_TOP_SRCDIR="@0@"'.format(meson.project_source_root()),
+  ],
+  include_directories: toplevel_inc,
+  dependencies: [
+    gobject,
+  ],
+)
+
+foreach test : tests
+  test_name = test.get('name')
+  test_srcs = test.get('srcs', test_name + '.c')
+  test_deps = test.get('deps', [])
+  test_incs = test.get('incs', [])
+  test_link = test.get('link', [])
+
+  test_exe = executable(
+    test_name,
+    test_srcs,
+    brush_settings_headers,
+    c_args: '-DLIBMYPAINT_TESTING_ABS_TOP_SRCDIR="@0@"'.format(meson.project_source_root()),
+    include_directories: [
+      toplevel_inc,
+      test_incs,
+    ],
+    link_with: [
+      libmypaint,
+      libmypaint_tests_lib,
+      test_link,
+    ],
+    dependencies: [
+      gobject,
+      libmath,
+      libprofiler,
+      test_deps,
+    ],
+  )
+
+  test(
+    test_name,
+    test_exe,
+    # test-gegl-surface takes very long without optimizations.
+    timeout: 0,
```

Comment:
Would it make sense to force optimizations then?

Reply:
Even then, it will take ~300 sec (IIRC) rather than ~500 sec.

Reply:
Maybe we should make the timeouts settable per test? There is one other that takes over 100 sec but most of them finish reasonably quickly.

Reply:
Sure, makes sense.

Even for the one that takes 500 seconds, I would suggest defining that or something near it as the upper bound, so meson can still abort things if it notices the test has taken 40 minutes and counting.

Reply:
It does not look like there is an upper bound. Each time I increase it and run it few times, I get some timeouts. Maybe it depends on how busy the CPU is but I encountered it taking more than 1000 seconds, even with optimizations.

Reply:
Yikes. :(

If it is too hard to figure out a reliable timeout indicator then sure, it makes sense to suppress detecting the timeout entirely.